br/pkg/task/restore.go

```diff
@@ -1166,6 +1173,66 @@ func runRestore(c context.Context, g glue.Glue, cmdName string, cfg *RestoreConf
 	return nil
 }
 
+func getMaxReplica(ctx context.Context, mgr *conn.Mgr) (int, error) {
+	resp, err := mgr.GetPDHTTPClient().GetReplicateConfig(ctx)
+	if err != nil {
+		return 0, errors.Trace(err)
+	}
+	key := "max-replicas"
+	val, ok := resp[key]
+	if !ok {
+		return 0, errors.Errorf("key %s not found in response %v", key, resp)
+	}
+	return int(val.(float64)), nil
+}
+
+func calNecessary(files []*backuppb.File, maxReplica int, storeCnt int) uint64 {
+	var totalSize uint64
+	for _, file := range files {
+		size := file.GetTotalBytes()
```

Comment:
Prefer use `file.Size` instead of `TotalBytes`, the former is the real size of the SST, which  is nearer to the size TiKV really uses . While the latter is the logical size (all key and values byte count), which will be greatly larger.